esmvaltool/cmorizers/obs/cmorize_obs_era_interim.py

```diff
@@ -0,0 +1,329 @@
+            if len(coord.points) > 1:
+                coord.guess_bounds()
+
+
+def _fix_frequency(cube, var):
+    # Here var_name is the CMIP name
+    # era-interim is in 3hr or 6hr or 12hr freq need to convert to daily
+    # only variables with step 12 need accounting time 00 AM as time 24 PM
+    if var['mip'] in {'day', 'Eday', 'CFday'}:
+        # accounting time 00 AM as time 24 PM
+        if cube.var_name in {'tasmax', 'tasmin', 'pr',
+                             'rsds', 'hfds', 'evspsbl',
+                             'rsdt', 'rss', 'prsn'}:
+            cube.coord('time').points = [
+                cube.coord('time').units.date2num(
+                    cell.point - timedelta(seconds=1)
+                )
+                for cell in cube.coord('time').cells()
+            ]
+        if cube.var_name == 'tasmax':
+            cube = daily_statistics(cube, 'max')
+        elif cube.var_name == 'tasmin':
+            cube = daily_statistics(cube, 'min')
+        elif cube.var_name in {'pr', 'rsds', 'hfds', 'evspsbl',
+                               'rsdt', 'rss', 'prsn'}:
+            # Sum is not available in daily_statistics so call iris directly
+            if not cube.coords('day_of_year'):
+                iris.coord_categorisation.add_day_of_year(cube, 'time')
+            if not cube.coords('year'):
+                iris.coord_categorisation.add_year(cube, 'time')
+            cube = cube.aggregated_by(['day_of_year', 'year'],
+                                      iris.analysis.SUM)
+        else:
+            cube = daily_statistics(cube, 'mean')
+        # Remove daily statistics helpers
+        cube.remove_coord(cube.coord('day_of_year'))
+        cube.remove_coord(cube.coord('year'))
+        # Correct the time bound
+        cube.coord('time').points = cube.coord('time').units.date2num(
+            [
+                cell.point.replace(hour=12, minute=0, second=0,
+                                   microsecond=0)
+                for cell in cube.coord('time').cells()
+            ]
+        )
+        cube.coord('time').bounds = None
+        cube.coord('time').guess_bounds()
+    return cube
+
+
+def _get_files(in_dir, var):
+    # Make a dictionary with the keys that are years
+    files_dict = defaultdict(list)
+    for in_file in var['files']:
+        files_lst = sorted(list(Path(in_dir).glob(in_file)))
+        for item in files_lst:
+            key = str(item.stem).split('_')[-1]
+            files_dict[key].append(item)
+    # Check if files are complete
+    for key in files_dict.copy():
+        if len(files_dict[key]) != len(var['files']):
+            logger.info("CMORizing %s at time '%s' needs '%s' input file/s",
+                        var['short_name'], key, len(var['files']))
+            files_dict.pop(key)
+    return files_dict
+
+
+def _extract_variable(in_file, var, cfg, out_dir):
+    if 'files' in var:
+        logger.info("CMORizing variable '%s' from input file/s '%s'",
```

Comment:
please try to avoid code duplication, there is no need to write the same log message code twice